### ISSUE TITLE
Android support?

### ISSUE THREAD

**Issue by shadmansaleh** [actboy168/bee.lua#14]
Can this be built on android or with NDK toolchain?

It is detected as a  platform that's why I was wondering if it was supported or not :D 
https://github.com/actboy168/bee.lua/blob/3c6764ad21d20cf279aea3077e823363f045e854/bee/platform.h#L14

I was trying to build this on [termux](https://github.com/termux/termux-app)

It mostly compiles but I get this error at the end
```
[57/59] build/linux/bin/bootstrap 3rd/bee.lua/test/test.lua 
     build/linux/bin/bootstrap: error loading module 'bee.platform' from file 'build/linux/bin/bee.so':
           dlopen failed: cannot locate symbol "_ZNKSt6__ndk14__fs10filesystem4path6__stemEv" referenced by "/data/data/com.termux/files/home/.cache/lsp/sumneko-lua/3rd/luamake/build/linux/bin/bee.so"...
           stack traceback:
           [C]: in ?
           [C]: in function 'require'  
           3rd/bee.lua/test/test.lua:19: in main chunk
           ...che/lsp/sumneko-lua/3rd/luamake/build/linux/bin/main.lua:24: in main chunk
           [C]: in ?
[58/59] Copy build/linux/bin/bee.so bee.so
```

Any idea what this might be ?

Also:
https://github.com/actboy168/bee.lua/blob/3c6764ad21d20cf279aea3077e823363f045e854/bee/platform.h#L59-L62 
2nd one is probably a typo and should be `BEE_CRT_VERSION` ?

**Comment by actboy168** [comment]
The low version of gcc needs -lstdc++fs to support filesystem.

**Comment by shadmansaleh** [comment]
> The low version of gcc needs -lstdc++fs to support filesystem.

clang-12 ? I'll try adding -lstdc++fs

**Comment by shadmansaleh** [comment]
I think stdc++fs is gcc specific . It's not available in clang .

**Comment by actboy168** [comment]
I have not tried to use it on Android.

**Comment by shadmansaleh** [comment]
Alright I'll let you know if I figure something out.

**Comment by actboy168** [comment]
filesystem requires NDK >= r22. I have fixed it in 9cf27392ead1dc02075625b5cefcdedc07dfa2f7.

**Comment by shadmansaleh** [comment]
Awesome . Just tested it now compiles 💯 . With a bit more hacking I've managed to get luamake and sumneko-lua-lsp some what working . Though for some reason clang -lstdc++ was failing with a link error about missing libc++_shared. But clang++ without -lstdc++ worked .
